books/RayTracingTheNextWeek.html

```diff
@@ -3815,34 +3815,41 @@
       public:
 
         bool hit(const ray& r, interval ray_t, hit_record& rec) const override {
-            // Change the ray from world space to object space
-            auto origin = r.origin();
-            auto direction = r.direction();
 
-            origin[0] = cos_theta*r.origin()[0] - sin_theta*r.origin()[2];
-            origin[2] = sin_theta*r.origin()[0] + cos_theta*r.origin()[2];
+            // Transform the ray from world space to object space.
 
-            direction[0] = cos_theta*r.direction()[0] - sin_theta*r.direction()[2];
-            direction[2] = sin_theta*r.direction()[0] + cos_theta*r.direction()[2];
+            auto origin = point3(
+                (cos_theta * r.origin().x()) - (sin_theta * r.origin().z()),
+                r.origin().y(),
+                (sin_theta * r.origin().x()) + (cos_theta * r.origin().z())
+            );
+
+            auto direction = vec3(
+                (cos_theta * r.direction().x()) - (sin_theta * r.direction().z()),
+                r.direction().y(),
+                (sin_theta * r.direction().x()) + (cos_theta * r.direction().z())
+            );
 
             ray rotated_r(origin, direction, r.time());
 
-            // Determine whether an intersection exists in object space (and if so, where)
+            // Determine whether an intersection exists in object space (and if so, where).
+
             if (!object->hit(rotated_r, ray_t, rec))
                 return false;
 
-            // Change the intersection point from object space to world space
-            auto p = rec.p;
-            p[0] =  cos_theta*rec.p[0] + sin_theta*rec.p[2];
-            p[2] = -sin_theta*rec.p[0] + cos_theta*rec.p[2];
+            // Transform the intersection from object space back to world space.
```

Comment:
Before, there was a comment about the intersection and one about the normal.  Now the one about the normal is gone. Should that just be integrated into this comment, or did you drop mention of the normal on purpose?

Reply:
I considered the normal to be part of the intersection data (which I why I didn't say "intersection _point_".

Reply:
Thanks for checking.